x-pack/plugins/maps/public/connected_components/mb_map/mb_map.tsx

```diff
@@ -72,6 +72,13 @@ export interface Props {
   onSingleValueTrigger?: (actionId: string, key: string, value: RawValue) => void;
   geoFields: GeoFieldWithIndex[];
   renderTooltipContent?: RenderToolTipContent;
+  setAreTilesLoaded: (layerId: string, areTilesLoaded: boolean) => void;
+}
+
+interface Tile {
```

Comment:
How about moving all of the tile loading logic into a separate component? For example, TooltipControl, isolates all of the listeners and logic for tooltip events. I think it would be nice to do the same for tile events.